src/AppRoot.tsx

```diff
@@ -312,36 +375,48 @@ class AppRoot extends React.Component<RouteComponentProps, State> {
         votes: this.getVotes(),
       })
     }
-    Mousetrap.bind(removeElectionShortcuts, this.unconfigure)
     document.addEventListener('keydown', handleGamepadKeyboardEvent)
     document.documentElement.setAttribute('data-useragent', navigator.userAgent)
     this.setDocumentFontSize()
-
-    const { signal } = this.machineIdAbortController
-    fetch('/machine-id', { signal })
-      .then(response => response.json())
-      .then(response => {
-        const { machineId } = response
-        if (machineId) {
-          this.setState({
-            machineId,
-          })
-        }
-      })
-      .catch(() => {
-        // TODO: what should happen if `machineId` is not returned?
-      })
-
+    this.setMachineId()
     this.startPolling()
   }
 
   public componentWillUnmount = /* istanbul ignore next - triggering keystrokes issue - https://github.com/votingworks/bmd/issues/62 */ () => {
     this.machineIdAbortController.abort()
-    Mousetrap.unbind(removeElectionShortcuts)
     document.removeEventListener('keydown', handleGamepadKeyboardEvent)
     this.stopPolling()
   }
 
+  public setMachineId = async () => {
+    const { signal } = this.machineIdAbortController
+    try {
+      const response = await fetch('/machine-id', { signal })
+      const { machineId }: MachineIdAPI = await response.json()
```

Comment:
There's a relatively new utility function I added that can shorten this.

```suggestion
      const { machineId } = await fetchJSON<MachineIdAPI>('/machine-id', { signal })
```

Reply:
Thx. Will implement.